File: implementation/scipamato/scipamato-public-persistence-jooq/src/test/java/ch/difty/scipamato/publ/persistence/keyword/JooqKeywordRepoIntegrationTest.java
package ch.difty.scipamato.publ.persistence.keyword;

import static org.assertj.core.api.Assertions.assertThat;

import java.util.List;

import lombok.extern.slf4j.Slf4j;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;

import ch.difty.scipamato.publ.entity.Keyword;
import ch.difty.scipamato.publ.persistence.JooqBaseIntegrationTest;

@Slf4j
public class JooqKeywordRepoIntegrationTest extends JooqBaseIntegrationTest {

    @Autowired
    private JooqKeywordRepo repo;

    @Test
    public void findingAllKeywordsInGerman() {
        List<Keyword> keywords = repo.findKeywords("de");
        assertThat(keywords)
            .extracting("name")
            .containsExactly("Aerosol", "Aktivität, eingeschränkte", "Allergie");
        keywords.forEach((c) -> log.debug(c.toString()));
    }

    @Test
    public void findingAllKeywords1InEnglish() {
        List<Keyword> keywords = repo.findKeywords("en");
        assertThat(keywords)
            .extracting("name")
            .containsExactly("Aerosol", "Allergies", "Restricted activity");
        keywords.forEach((c) -> log.debug(c.toString()));
    }

    @Test
    public void findingAllKeywords1InFrench() {
        List<Keyword> keywords = repo.findKeywords("fr");
        assertThat(keywords)
            .extracting("name")
            .containsExactly("Activités réduites", "Aérosol", "Allergie");
        keywords.forEach((c) -> log.debug(c.toString()));
    }

}

File: public/public-persistence-jooq/src/integration-test/kotlin/ch/difty/scipamato/publ/persistence/JooqKeywordRepoIntegrationTest.kt
package ch.difty.scipamato.publ.persistence

import ch.difty.scipamato.common.logger
import ch.difty.scipamato.publ.persistence.keyword.JooqKeywordRepo
import org.amshove.kluent.shouldContainAll
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.jooq.JooqTest
import org.testcontainers.junit.jupiter.Testcontainers

private val log = logger()

@JooqTest
@Testcontainers
internal open class JooqKeywordRepoIntegrationTest {

    @Autowired
    private lateinit var repo: JooqKeywordRepo

    @Test
    @Suppress("SpellCheckingInspection")
    fun findingAllKeywordsInGerman() {
        val keywords = repo.findKeywords("de")
        keywords.map { it.name } shouldContainAll listOf("Aerosol", "Aktivität, eingeschränkte", "Allergie")
        keywords.forEach { c -> log.debug(c.toString()) }
    }

    @Test
    fun findingAllKeywords1InEnglish() {
        val keywords = repo.findKeywords("en")
        keywords.map { it.name } shouldContainAll listOf("Aerosol", "Allergies", "Restricted activity")
        keywords.forEach { c -> log.debug(c.toString()) }
    }

    @Suppress("SpellCheckingInspection")
    @Test
    fun findingAllKeywords1InFrench() {
        @Suppress("SpellCheckingInspection")
        val keywords = repo.findKeywords("fr")
        keywords.map { it.name } shouldContainAll listOf("Activités réduites", "Aérosol", "Allergie")
        keywords.forEach { c -> log.debug(c.toString()) }
    }
}
